src/main/java/ru/testtask/converter/DTOConverter.java

```diff
@@ -0,0 +1,49 @@
+package ru.testtask.converter;
+
+import lombok.NoArgsConstructor;
+import org.springframework.stereotype.Component;
+import ru.testtask.dto.AttrDTO;
+import ru.testtask.dto.ClientProjectDTO;
+import ru.testtask.dto.ProjectDTO;
+import ru.testtask.model.Attribute;
+import ru.testtask.model.Project;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Component
+@NoArgsConstructor
+public class DTOConverter {
+
+    public ProjectDTO convertToDTO(Project project){
+        List<AttrDTO> attrs = getDTOAttrList(project.getAttributes());
+        return ProjectDTO.builder().id(project.getId()).name(project.getName()).attrs(attrs).build();
```

Comment:
You can use ModelMapper to map one object to different. And you can configure one if some fiels has a different names and types